src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/retriever/arcaddon/arc_add_on_retriever.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+from ...entity.CustomerResource import CustomerResource
+from ...entity._credentials import Credentials
+from ...exception import DataNotFoundException
+from ...executor.azcli._AzCliExecutor import AzCliExecutor
+from ...constants import Constant
+from ...entity.AzCli import AzCli
+from ...retriever._retriever import Retriever
+
+
+class ArcAddOnRetriever(Retriever):
+    instance = None
+    _arc_add_on_url = Constant.MGMT_URL + "addons" + "?" + Constant.API_VERSION + "=" + Constant.PREVIEW_API_VERSION_VALUE
+    
+    def __new__(cls):
+        if cls.instance is None:
+            cls._az_cli_executor = AzCliExecutor()
+            cls.instance = object.__new__(cls)
+        return cls.instance
+
+    def retrieve_data(self, customer_resource):
+        customer_res: CustomerResource = customer_resource
+        arc_add_on_url = self._arc_add_on_url.format(customer_res.subscription_id,
```

Comment:
same, this is addons url, not specific to just arc addon

Reply:
changed to addon url